### PR TITLE
Data migration + testing

### PR DESCRIPTION
**Pull request by koenvanderveen**
## Description

Includes unittest + full data migration from 0.8.6 to current dev.
- I did not include the k8s test for now in CI, but it works locally. Once we have a k8s version to test against, it makes sense to include it in CI, right now it doesnt add a lot of value
- The k8s test still has a few small issues that I commented out, we can tackle that for next PR

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: packages/syft/src/syft/service/action/action_service.py [hunk 1]
```diff
@@ -572,8 +574,11 @@ def store_permission(
 
         def blob_permission(
             x: SyftVerifyKey | None = None,
-        ) -> ActionObjectPermission:
-            return ActionObjectPermission(result_blob_id, read_permission, x)
+        ) -> ActionObjectPermission | None:
```
Comment (koenvanderveen) [inline packages/syft/src/syft/service/action/action_service.py:577]: how does this work?

File: packages/syft/src/syft/service/migration/migration_service.py [hunk 2]
```diff
@@ -0,0 +1,668 @@
...
+        path="migration.get_all_store_metadata",
+        name="get_all_store_metadata",
+        roles=ADMIN_ROLE_LEVEL,
+    )
+    def get_all_store_metadata(
+        self,
+        context: AuthedServiceContext,
+        document_store_object_types: list[type[SyftObject]] | None = None,
+        include_action_store: bool = True,
+    ) -> dict[str, StoreMetadata] | SyftError:
```
Comment (koenvanderveen) [inline packages/syft/src/syft/service/migration/migration_service.py:133]: maybe rename to something more permissiony

File: packages/syft/src/syft/store/kv_document_store.py [hunk 3]
```diff
@@ -422,6 +429,7 @@ def _update(
         obj: SyftObject,
         has_permission: bool = False,
         overwrite: bool = False,
+        allow_missing_keys: bool = False,
```
Comment (koenvanderveen) [inline packages/syft/src/syft/store/kv_document_store.py:432]: check if this is implemented on both

File: packages/syft/src/syft/store/kv_document_store.py [hunk 4]
```diff
@@ -476,6 +495,11 @@ def _update(
                 return Err(f"Failed to update obj {obj}, you have no permission")
 
         except Exception as e:
+            # third party
```
Comment (koenvanderveen) [inline packages/syft/src/syft/store/kv_document_store.py:498]: remove

File: packages/syft/src/syft/store/kv_document_store.py [hunk 5]
```diff
@@ -680,14 +704,21 @@ def _migrate_data(
                 try:
                     migrated_value = value.migrate_to(to_klass.__version__, context)
                 except Exception:
-                    return Err(f"Failed to migrate data to {to_klass} for qk: {key}")
+                    # stdlib
+                    import traceback
+
```
Comment (koenvanderveen) [inline packages/syft/src/syft/store/kv_document_store.py:709]: remove

File: packages/syft/src/syft/util/schema.py [hunk 6]
```diff
@@ -211,6 +211,7 @@ def resolve_references(json_mappings: dict[str, dict]) -> dict[str, dict]:
 
 
 def generate_json_schemas(output_path: str | None = None) -> None:
+    # TODO: should we also replace this with the SyftObjectRegistry?
```
Comment (koenvanderveen) [inline packages/syft/src/syft/util/schema.py:214]: todo

**Review by eelcovdw — APPROVED** [review]

**Pull request merged**